src/test/java/com/google/cloud/spanner/jdbc/JdbcStatementTest.java

```diff
@@ -353,7 +412,24 @@ public void testExecuteUpdateWithSelectStatement() {
     } catch (SQLException e) {
       assertThat(
               JdbcExceptionMatcher.matchCodeAndMessage(
-                      Code.INVALID_ARGUMENT, "The statement is not an update or DDL statement")
+                      Code.INVALID_ARGUMENT,
+                      "The statement is not a normal update or DDL statement")
+                  .matches(e))
+          .isTrue();
+    }
+  }
+
+  @Test
+  public void testExecuteUpdateWithDmlReturningStatement() {
+    try {
+      Statement statement = createStatement();
+      statement.executeUpdate(getDmlReturningSql());
+      fail("missing expected exception");
```

Comment:
nit: use `assertThrows`, even if all the other tests in the file use `fail(..)`. Preferably, we should try to clean up these `try-fail` constructs when we have time